runtime/browser/xwalk_presentation_service_helper.h

```diff
@@ -0,0 +1,265 @@
+// Copyright (c) 2016 Intel Corporation. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_XWALK_PRESENTATION_SERVICE_HELPER_H_
+#define XWALK_RUNTIME_BROWSER_XWALK_PRESENTATION_SERVICE_HELPER_H_
+
+#include <string>
+#include <vector>
+
+#include "base/guid.h"
+#include "base/logging.h"
+#include "base/memory/ref_counted.h"
+#include "base/memory/singleton.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/presentation_screen_availability_listener.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/render_process_host.h"
+#include "xwalk/application/browser/application_system.h"
+#include "xwalk/application/browser/application_service.h"
+#include "xwalk/runtime/browser/runtime.h"
+#include "xwalk/runtime/browser/xwalk_browser_context.h"
+#include "xwalk/runtime/browser/xwalk_runner.h"
+
+#include "xwalk/runtime/browser/xwalk_presentation_service_delegate.h"
+
+#if defined(OS_ANDROID)
+#include "xwalk/runtime/browser/android/xwalk_presentation_host.h"
+#elif defined(OS_WIN)
+#include <Windows.h>
+#endif
+
+namespace xwalk {
+
+using application::Application;
+using application::ApplicationService;
+
+using content::PresentationScreenAvailabilityListener;
+using content::PresentationSessionMessage;
+using content::RenderFrameHost;
+using content::PresentationConnectionStateChangedCallback;
+
+using DelegateObserver = content::PresentationServiceDelegate::Observer;
+using PresentationSessionErrorCallback =
+    content::PresentationSessionErrorCallback;
+using PresentationSessionStartedCallback =
+    content::PresentationSessionStartedCallback;
+using RenderFrameHostId =
+    XWalkPresentationServiceDelegate::RenderFrameHostId;
+using SessionInfo = content::PresentationSessionInfo;
+
+#if defined(OS_ANDROID)
+using SystemString = std::basic_string<char>;
+#elif defined(OS_WIN)
+using SystemString = std::basic_string<TCHAR>;
+#endif
```

Comment:
I think this is also a cause to break Linux build, we'd better to use more generic way.